clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkThread.java

```diff
@@ -145,6 +154,7 @@ void runOnce() {
                 .map(networkClientDelegate::addAll)
                 .reduce(MAX_POLL_TIMEOUT_MS, Math::min);
         networkClientDelegate.poll(pollWaitTimeMs, currentTimeMs);
+        maybePropagateMetadataError();
```

Comment:
Any reason to have this here instead of inside the `NetworkClientDelegate` poll? 

I would expect it inside the NetworkClientDelegate poll, which would be the equivalent of the ConsumerNetworkClient (where the legacy consumer has this same logic). I know here we are calling them sequentially so we'll get the same result, but networkClientDelegate.poll is called from other places too (on close), so we wouldn't propagate the error on those cases I guess.

Reply:
hey @lianetm - it seems like invalidTopicException is only thrown in the `poll()`.  i guess `InvalidTopicException` is irrelevant during closing as the client is shutting down anyway.

Reply:
I looked at `ConsumerNetworkThread.run()` because `Metadata.fatalException` is now propagated along with `InvalidTopicException` due to this change.
When I looked at the code, there was the code where `networkClientDelegate.poll` was actually called separately in addition to `ConsumerNetworkThread.runOnce()`.
```
    // ConsumerNetworkThread.run()
    public void run() {
        try {
            ...
            while (running) {
                ...
                runOnce();
                ...
            }
        } finally {
            // call networkClientDelegate.poll()
            cleanup();
        }
    }
```
Therefore, I thought that the Metadata error might be missed at least once when the `AsyncKafkaConsumer.close()` is called and there are a Metadata error.
So I applied it to the `NetworkClientDelegate`.

Let me know if I've misunderstood